src/bits/shared.h

```diff
@@ -53,6 +53,19 @@ namespace detail {
     template <typename FN, typename... ARGS>
     using invoke_result_t = typename invoke_result<FN, ARGS...>::type;
 
+    // Test if two variables is at the same address.
+    template <typename T, typename S>
+    bool is_at_same_address(T, S)
+    {
+        return false;
+    }
+
+    template <typename T>
+    bool is_at_same_address(T &&x, T &&y)
+    {
+        return x == y;
+    }
```

Comment:
Uhm... what is this supposed to do? This is comparing objects, not pointers

Reply:
right, my bad. I was trying to check if it x and y is the same container.